src/v/datalake/translation/partition_translator.cc

```diff
@@ -325,6 +321,9 @@ partition_translator::do_translate_once(retry_chain_node& parent_rcn) {
             max_translated_offset = last_translated_offset;
             result = translation_success::yes;
         }
+        if (last_translated_offset < read_end_offset) {
```

Comment:
this looks fine but I think there are a couple of (minor) issues.

* max_offset_for_translation() may have advanced as the translation is trying to catchup in which case we will still incur an extra sleep at the end of this do_translate_once() (once it returns success)
* there is still an exponential backoff in increments of 300ms doing this way, so still some additional sleeps

A temporary alternative  may be something like

> if (has_enough_data_to_translate()) {
>     // continue translation
> } else {
>      co_await ss::sleep_abortable(_jitter.next_duration(), _as); 
> 
> }

I have some ideas to revamp this code altogether.

Reply:
Yeah I figured it's worth pinning max_offset_for_translation() to a specific value per "round", and doing a sleep even if it moves in case it led to some kind of starvation. But this isn't a tight loop anyway, so maybe you're right

Agreed otherwise

Reply:
Updated